r-package/grf/tests/testthat/test_regression_forest.R

```diff
@@ -113,27 +113,54 @@ test_that("regression forest tuning only cross-validates null parameters", {
 test_that("local linear prediction gives reasonable estimates", {
     f = function(x){x[1] + 2*x[2] + 2*x[3]**2}
     n = 600
-    p = 4
+    p = 5
     X = matrix(rnorm(n*p), n, p)
     MU = apply(X, FUN=f, MARGIN=1)
     Y = MU + rnorm(n)
 
     forest = regression_forest(X, Y)
     preds.grf.oob = predict(forest)
-    preds.ll.oob = predict(forest, local.linear=TRUE)
-    mse.grf.oob = mean( (preds.grf.oob$predictions - MU)^2 )
-    mse.ll.oob = mean( (preds.ll.oob$predictions - MU)^2 )
+    preds.ll.oob = predict(forest, linear.correction.variables = 1:p, lambda = 0)
 
-  expect_true( mse.ll.oob < 1.5 )
-  expect_true( mse.ll.oob < mse.grf.oob / 1.5 )
+    mse.grf.oob = mean((preds.grf.oob$predictions - MU)^2)
+    mse.ll.oob = mean((preds.ll.oob$predictions - MU)^2)
+
+    expect_true(mse.ll.oob < 1)
+    expect_true(mse.ll.oob < mse.grf.oob / 2)
 
     X.test = matrix(rnorm(n*p), n, p)
     MU.test = apply(X.test, FUN=f, MARGIN=1)
+
     preds.grf = predict(forest, X.test)
-    preds.ll = predict(forest, X.test, local.linear=TRUE)
-    mse.grf = mean( (preds.grf$predictions - MU.test)^2 )
-    mse.ll = mean( (preds.ll$predictions - MU.test)^2 )
+    preds.ll = predict(forest, X.test, linear.correction.variables = 1:p, lambda=0.1)
+
+    mse.grf = mean((preds.grf$predictions - MU.test)^2)
+    mse.ll = mean((preds.ll$predictions - MU.test)^2)
+
+    expect_true(mse.ll < 1)
+    expect_true(mse.ll < mse.grf / 1.5)
+})
+
+test_that("linear correction variables function as expected", {
+    f = function(x){x[1] + 2*x[2] + 2*x[3]**2}
+    n = 400
+    p = 20
+    X = matrix(rnorm(n*p), n, p)
+    MU = apply(X, FUN=f, MARGIN=1)
+    Y = MU + rnorm(n)
+
+    forest = regression_forest(X, Y)
+    preds = predict(forest, linear.correction.variables = 1:20)
+    mse = mean((preds$predictions - MU)^2)
+
+    preds.selected = predict(forest, linear.correction.variables = 1:3)
+    mse.selected = mean((preds.selected$predictions - MU)^2)
+
+    expect_true(mse.selected < mse / 1.5)
+
+    preds.automatic = predict(forest, select.correction.variables = TRUE)
```

Comment:
I think you need to remove these lines, as `select.correction.variables` is no longer an option.